File: shopper-web/src/app/model/ListItem.ts
import {Product} from "./Product";
import {Category} from "./Category";

export class ListItem {
    id?: number;
    product: Product;
    name?: string;
    description?: string;
    quantity?: number;
    unit?: string;
    category?: Category;
    done?: boolean;

    displayName() {
        return this.name ? this.name : this.product.name;
    }
}

File: shopper-watch-companion/app/src/main/java/pl/qprogramming/shopper/watch/model/ListItem.java
package pl.qprogramming.shopper.watch.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class ListItem {
    private Long id;
    private Product product;
    private String name;
    private String description;
    private float quantity;
    private String unit;
    private Category category;
    private boolean done;
}